### ISSUE TITLE
HttpClient hangs when logging level in DEBUG

### ISSUE THREAD

**Issue by megaza** [eclipse/jetty.project#1419]
Used HttpClient with SslContextFactory and `-Dorg.eclipse.jetty.LEVEL=DEBUG`.

Hangs on GET and throws:
```
2017-03-23 12:42:28.717:DBUG:oejut.QueuedThreadPool:HttpClient@885284298-10: ran org.eclipse.jetty.io.ManagedSelector$$Lambda$5/597255128@3abbfa04
java.util.concurrent.ExecutionException: java.lang.NullPointerException
	at org.eclipse.jetty.client.util.FutureResponseListener.getResult(FutureResponseListener.java:118)
	at org.eclipse.jetty.client.util.FutureResponseListener.get(FutureResponseListener.java:101)
	at org.eclipse.jetty.client.HttpRequest.send(HttpRequest.java:652)
	at org.eclipse.jetty.client.HttpClient.GET(HttpClient.java:342)
	at org.eclipse.jetty.client.HttpClient.GET(HttpClient.java:327)
	at com.esq.rmm.plugin.jetty9client.Application.main(Application.java:79)
Caused by: java.lang.NullPointerException
	at org.eclipse.jetty.io.ssl.SslConnection.toConnectionString(SslConnection.java:315)
	at org.eclipse.jetty.io.AbstractEndPoint.toConnectionString(AbstractEndPoint.java:469)
	at org.eclipse.jetty.io.AbstractEndPoint.toString(AbstractEndPoint.java:438)
	at org.eclipse.jetty.io.ssl.SslConnection$DecryptedEndPoint.toString(SslConnection.java:1141)
	at java.util.Formatter$FormatSpecifier.printString(Formatter.java:2886)
	at java.util.Formatter$FormatSpecifier.print(Formatter.java:2763)
	at java.util.Formatter.format(Formatter.java:2520)
	at java.util.Formatter.format(Formatter.java:2455)
	at java.lang.String.format(String.java:2940)
	at org.eclipse.jetty.io.AbstractConnection.toString(AbstractConnection.java:264)
	at java.lang.String.valueOf(String.java:2994)
	at org.eclipse.jetty.util.log.StdErrLog.format(StdErrLog.java:553)
	at org.eclipse.jetty.util.log.StdErrLog.format(StdErrLog.java:442)
	at org.eclipse.jetty.util.log.StdErrLog.debug(StdErrLog.java:406)
	at org.eclipse.jetty.client.http.HttpClientTransportOverHTTP.newConnection(HttpClientTransportOverHTTP.java:59)
	at org.eclipse.jetty.io.ssl.SslClientConnectionFactory.newConnection(SslClientConnectionFactory.java:67)
	at org.eclipse.jetty.client.AbstractHttpClientTransport$ClientSelectorManager.newConnection(AbstractHttpClientTransport.java:191)
	at org.eclipse.jetty.io.ManagedSelector.createEndPoint(ManagedSelector.java:488)
	at org.eclipse.jetty.io.ManagedSelector.access$1700(ManagedSelector.java:60)
	at org.eclipse.jetty.io.ManagedSelector$CreateEndPoint.run(ManagedSelector.java:667)
	at org.eclipse.jetty.util.thread.Invocable.invokePreferred(Invocable.java:122)
	at org.eclipse.jetty.util.thread.strategy.ExecutingExecutionStrategy.invoke(ExecutingExecutionStrategy.java:58)
	at org.eclipse.jetty.util.thread.strategy.ExecuteProduceConsume.produceConsume(ExecuteProduceConsume.java:201)
	at org.eclipse.jetty.util.thread.strategy.ExecuteProduceConsume.produce(ExecuteProduceConsume.java:97)
	at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:672)
	at org.eclipse.jetty.util.thread.QueuedThreadPool$2.run(QueuedThreadPool.java:590)
	at java.lang.Thread.run(Thread.java:745)
```

**Comment by WalkerWatch** [comment]
What version of Jetty are you using? I am trying to replicate this.

**Comment by megaza** [comment]
jetty-client: 9.4.2.v20170220.

Bug was introduced in 9.4 RC3 by changing SslConnection.toConnectionString()
Seems that debug log is not enabled in unit test.

Tested simple GET request to https://www.google.com (with keystore).

**Comment by WalkerWatch** [comment]
@gregw Can you take a  look at this? Change was made in regards to #572

**Comment by joakime** [comment]
This was fixed in commit 7547fc4b2951a8f370ab1f5c74680bee2908fd3d
